modules/031-linstor/images/linstor-pools-importer/main.go

```diff
@@ -0,0 +1,481 @@
+	storagev1 "k8s.io/api/storage/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog"
+
+	lclient "github.com/LINBIT/golinstor/client"
+)
+
+const (
+	lvmConfig      = `devices {filter=["r|^/dev/drbd*|"]}`
+	linstorPrefix  = "linstor"
+	maxReplicasNum = 3
+)
+
+var (
+	nodeName  = os.Getenv("NODE_NAME")
+	podName   = os.Getenv("POD_NAME")
+	namespace = os.Getenv("NAMESPACE")
+)
+
+var supportedProviderKinds = []lclient.ProviderKind{lclient.LVM, lclient.LVM_THIN}
+
+func printVersion() {
+	klog.Info(fmt.Sprintf("Go Version: %s", runtime.Version()))
+	klog.Info(fmt.Sprintf("Go OS/Arch: %s/%s", runtime.GOOS, runtime.GOARCH))
+}
+
+func main() {
+
+	delaySeconds := flag.Int("delay", 10, "Delay in seconds between scanning attempts")
+
+	flag.Parse()
+	printVersion()
+
+	clientConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+		clientcmd.NewDefaultClientConfigLoadingRules(),
+		&clientcmd.ConfigOverrides{},
+	)
+
+	if nodeName == "" {
+		klog.Fatalln("Required NODE_NAME env variable is not specified!")
+	}
+	klog.Infof("nodeName: %+v", nodeName)
+	if podName == "" {
+		klog.Fatalln("Required POD_NAME env variable is not specified!")
+	}
+	klog.Infof("podName: %+v", podName)
+	if namespace == "" {
+		klog.Fatalln("Required NAMESPACE env variable is not specified!")
+	}
+	klog.Infof("namespace: %+v", namespace)
+
+	ctx := context.TODO()
+
+	// Get a config to talk to the apiserver
+	config, err := clientConfig.ClientConfig()
+	if err != nil {
+		klog.Errorln("Failed to get kubernetes config:", err)
+	}
+
+	kc, err := kclient.New(config, kclient.Options{})
+	if err != nil {
+		klog.Fatalln("Failed to create Kubernetes client:", err)
+	}
+
+	var pod v1.Pod
+	err = kc.Get(ctx, types.NamespacedName{Name: podName, Namespace: namespace}, &pod)
+	if err != nil {
+		klog.Fatalf("look up owner(s) of pod: %v", err)
+	}
+	owner := v1.ObjectReference{
+		APIVersion: "v1",
+		Kind:       "Pod",
+		Name:       pod.GetName(),
+		Namespace:  pod.GetNamespace(),
+		UID:        pod.GetUID(),
+	}
+	klog.Infof("using %s/%s as owner for Kubernetes events", owner.Kind, owner.Name)
+
+	lc, err := lclient.NewClient()
+	if err != nil {
+		klog.Fatalln("failed to create LINSTOR client:", err)
+	}
+
+	klog.Infof("Starting main loop (delay: %d seconds)", *delaySeconds)
+	candidatesChannel := runCandidatesLoop(ctx, getCandidates, time.Duration(*delaySeconds)*time.Second)
+
+	for candidate := range candidatesChannel {
+		klog.Infof("Got %s candidate: %+v\n", candidate.GetProviderKind(), candidate)
+		// Create storage pool in LINSTOR
+		storagePool, err := makeLinstorStoragePool(candidate)
+		if err != nil {
+			klog.Fatalln("failed to generate LINSTOR storage pool:", err)
+		}
+		_, err = lc.Nodes.Get(ctx, nodeName)
+		if err != nil {
+			klog.Fatalln("Failed to get LINSTOR node", err)
+		}
+		_, err = lc.Nodes.GetStoragePool(ctx, nodeName, storagePool.StoragePoolName)
+		if err != nil { // TODO check for 404
```

Comment:
404 is fine, as it means that we need to create a storagepool.
I skipped the implementation of check for other API errors because it can fail later anyway.